src/MoBi.Presentation/MenusAndBars/ContextMenus/ContextMenuForModule.cs

```diff
@@ -43,10 +43,20 @@ public IContextMenu InitializeWith(ObjectBaseDTO dto, IPresenter presenter)
          _allMenuItems.Add(createSaveItemFor(module));
          _allMenuItems.Add(createRemoveItemFor(module));
          _allMenuItems.Add(createCloneMenuItem(module));
+
+         if(module.HasSnapshot)
+            _allMenuItems.Add(createRecreateMenuItem(module));
          
```

Comment:
Should we additionally check that the module is a PK-Sim module?
(Thinking about a scenario where we convert a PK-Sim Module to an Extension module)

Reply:
I'm not sure about that. You have a module with an snapshot, you can recreate the original module, and it will be different from the one containing the snapshot. I think it's ok. In this iteration we are just loading the new module along with the existing one.

Reply:
agreed. It will create a new entry rightr?

Reply:
Yeah, a new one is created